graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/searchtypes/pivot/buckets/OSValuesHandler.java

```diff
@@ -101,6 +105,30 @@ private Script scriptForPivots(Collection<String> pivots) {
         return new Script(scriptSource);
     }
 
+    private TermsAggregationBuilder applyOrdering(Pivot pivot, TermsAggregationBuilder terms, List<BucketOrder> ordering, OSGeneratedQueryContext queryContext) {
+        return sortsOnNumericPivotField(pivot, queryContext)
+                .map(pivotSort -> terms
+                        .subAggregation(AggregationBuilders.max(SORT_HELPER).field(pivotSort.field()))
+                        .order(BucketOrder.aggregation(SORT_HELPER, SortSpec.Direction.Ascending.equals(pivotSort.direction()))))
+                .orElseGet(() -> terms
+                        .order(ordering.isEmpty() ? List.of(DEFAULT_ORDER) : ordering));
+    }
+
+    private Optional<PivotSort> sortsOnNumericPivotField(Pivot pivot, OSGeneratedQueryContext queryContext) {
+        return Optional.ofNullable(pivot.sort())
+                .filter(sorts -> sorts.size() == 1)
+                .map(sorts -> sorts.get(0))
+                .filter(sort -> sort instanceof PivotSort)
+                .map(sort -> (PivotSort) sort)
+                .filter(pivotSort -> queryContext.fieldType(pivot.effectiveStreams(), pivotSort.field())
```

Comment:
I have encountered a situation when `pivot.effectiveStreams()` returns empty collection, which makes this method return empty optional and results in a situation when you do not have numeric sort applied...

![Uploading image.png…]()

Reply:
I have fixed this by using `Query#effectiveStreams` instead.